dts/bindings/adc/gd,gd32-adc.yaml

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2021 BrainCo Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: GigaDevice GD32 ADC
+
+compatible: "gd,gd32-adc"
+
+include: [adc-controller.yaml, pinctrl-device.yaml]
+
+properties:
+    reg:
+      required: true
+
+    rcu-periph-clock:
+      type: int
+      description: Reset Control Unit Peripheral Clock ID
+      required: true
+
+    rcu-clock-source:
+      type: int
+      required: false
+      description: |
+        Some ADC have additional clock source, like IRC14M or IRC28M.
+        RCU(Reset Control Unit) HAL provide a enum called rcu_adc_clock_enum to
+        select the ADC clock source.
+
+        This use an integer to mapping the rcu_adc_clock_enum, and may have
+        different meaning between GD32 SoCs. For example, GD32F3X0 RCU
+        rcu_adc_clock_enum mapping as below:
+          - 0 RCU_ADCCK_IRC28M_DIV2
+          - 1 RCU_ADCCK_IRC28M
+          - 2 RCU_ADCCK_APB2_DIV2
+          - 3 RCU_ADCCK_AHB_DIV3
+          - 4 RCU_ADCCK_APB2_DIV4
+          - 5 RCU_ADCCK_AHB_DIV5
+          - 6 RCU_ADCCK_APB2_DIV6
+          - 7 RCU_ADCCK_AHB_DIV7
+          - 8 RCU_ADCCK_APB2_DIV8
+          - 9 RCU_ADCCK_AHB_DIV9
+      enum:
+        - 0
+        - 1
+        - 2
+        - 3
+        - 4
+        - 5
+        - 6
+        - 7
+        - 8
+        - 9
```

Comment:
better provide definitions in `include/dts-bindings/`, see this out of tree example: https://github.com/teslabs/spinner/blob/main/include/dts-bindings/adc/stm32f3xx.h.

Reply:
Thanks for you advice!

Reply:
I add a new commit to hold this. If everything is OK, it will be merge to old commit.